### PR TITLE
MINOR: improve JavaDocs about global state stores

### PR DESCRIPTION
**Pull request by mjsax**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @bbejeck @vvcephei @ableegoldman

**Review by guozhangwang — APPROVED** [review]

**Comment by mjsax** [comment]
Test result is not available any longer. Java8 failed. Java11 passed.

Retest this please.

Call for second review anyof @bbejeck @vvcephei @ableegoldman

**Review by ableegoldman — APPROVED** [review]
LGTM

**Review by bbejeck — APPROVED** [review]
LGTM

**Comment by bbejeck** [comment]
Waiting for Jenkins to pass for merging

**Comment by mjsax** [comment]
Just realized, that ScalaDocs should get the same improvement... add a commit.

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/StreamsBuilder.scala [hunk 1]
```diff
@@ -77,7 +78,7 @@ class StreamsBuilder(inner: StreamsBuilderJ = new StreamsBuilderJ) {
   /**
    * Create a [[kstream.KStream]] from the specified topic pattern.
    *
-   * @param topics the topic name pattern
+   * @param topicPattern the topic name pattern
```
Comment (mjsax) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/StreamsBuilder.scala:81]: fix on the side

File: streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/StreamsBuilder.scala [hunk 2]
```diff
@@ -27,6 +27,7 @@ import org.apache.kafka.streams.state.StoreBuilder
 import org.apache.kafka.streams.{Topology, StreamsBuilder => StreamsBuilderJ}
 import org.apache.kafka.streams.scala.kstream._
 import ImplicitConversions._
+import org.apache.kafka.streams.errors.TopologyException
```
Comment (mjsax) [inline streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/StreamsBuilder.scala:30]: fix on the side

**Comment by bbejeck** [comment]
@mjsax failures seem related (`spotlessScala` format related)

**Comment by mjsax** [comment]
This should fix it...

**Comment by bbejeck** [comment]
Merged #6359 into trunk

**Pull request merged**